idx/memory/memory.go

```diff
@@ -1411,8 +1441,8 @@ ORGS:
 			m.Unlock()
 			tl.Add(time.Since(lockStart))
 			pruned = append(pruned, defs...)
-
 		}
+		m.findCache.Purge(org)
```

Comment:
similar like in the case of `.Delete()` this could be

```suggestion
		if len(pruned) {
		    m.findCache.Purge(org)
		}
```

Reply:
that is already the case.  The loop exists early if `len(paths) == 0`